### PR TITLE
[cleanup][broker] PIP 45: Deprecate zookeeper settings

### PR DESCRIPTION
**Pull request by mattisonchao**
### Motivation

To deprecate the zookeeper settings. PIP 45 introduced a unified metadata store API for Pulsar,
so we should make the metadata-related configuration start with `metadataStore`, not `zooKeeper`/`ZK`.

### Modifications

- Change `managedLedgerMaxUnackedRangesToPersistInZooKeeper` to `managedLedgerMaxUnackedRangesToPersistInMetadataStore`
- Change ML config `maxUnackedRangesToPersistInZk` to `maxUnackedRangesToPersistInMetadataStore`.

### Verifying this change

- [x] Make sure that the change passes the CI checks.

### Does this pull request potentially affect one of the following parts:

  - The default values of configurations: (yes)

### Documentation

- [x] `doc-required` 
(Your PR needs to update docs and you will update later)

### Compatibility

Introduce `getManagedLedgerMaxUnackedRangesToPersistInMetadataStore` method to compatible with previous configurations.

### CODE REVIEW

**Review by HQebupt — APPROVED** [review]

**Review by Jason918 — COMMENTED** [review]
Need to update `broker.conf`, too.

**Comment by mattisonchao** [comment]
> Need to update `broker.conf`, too.

Fixed, thanks to you.

File: conf/broker.conf [hunk 1]
```diff
@@ -1016,9 +1016,15 @@ managedLedgerCursorRolloverTimeInSeconds=14400
...
 
+# Max number of "acknowledgment holes" that can be stored in MetadataStore. If number of unack message range is higher
+# than this limit then broker will persist unacked ranges into bookkeeper to avoid additional data overhead into
+# MetadataStore.
+managedLedgerMaxUnackedRangesToPersistInMetadataStore=1000
+
 # Max number of "acknowledgment holes" that can be stored in Zookeeper. If number of unack message range is higher
 # than this limit then broker will persist unacked ranges into bookkeeper to avoid additional data overhead into
 # zookeeper.
+# Deprecated: use metadataStoreSessionTimeoutMillis
```
Comment (Jason918) [inline conf/broker.conf]: Please move this to the bottom of the file.

**Review by Jason918 — APPROVED** [review]
LGTM

**Review by nodece — APPROVED** [review]

**Review by shibd — APPROVED** [review]
LGTM

**Review by codelipenghui — APPROVED** [review]

**Pull request merged**